lib/home_widget.dart

```diff
@@ -195,13 +195,18 @@ class HomeWidget {
               'Tests should always use default Path provider for easier mocking',
             );
           }
-          final PathProviderFoundation provider = PathProviderFoundation();
-          directory = await provider.getContainerPath(
-            appGroupIdentifier: HomeWidget.groupId!,
-          );
+          if (Platform.isIOS) {
+            final PathProviderFoundation provider = PathProviderFoundation();
+            directory = await provider.getContainerPath(
+              appGroupIdentifier: HomeWidget.groupId!,
+            );
+          }
+          if (Platform.isAndroid) {
+            directory = (await getApplicationSupportDirectory()).path;
+          }
           // coverage:ignore-end
-        } on UnsupportedError catch (_) {
-          directory = (await getApplicationSupportDirectory()).path;
```

Comment:
I had this in the catch block as this enabled doing a golden test to test the capturing done by `renderFlutterWidget`. 
I have the following suggestion that should still make it work in the test but not crash:

```dart
if (Platform.isIOS) {
  try {
    final PathProviderFoundation ...
    assert(HomeWidget.groupId != null, 'No groupId defined. Did you forget to call `HomeWidget.setAppGroupId`');
   // ... Set directory
  } on UnsupportedError catch (e) {
    throw e
  }
} else {
  // Android Case
  directory = (await getApplicationSupportDirectory()).path;
}
```

That should basically have the same outcome as your suggestion but would allow the test to pass. Also I think adding the assertion with the hint for iOS could be really helpful as well

Reply:
How about this approach:

```dart
          if (Platform.isIOS) {
            if (HomeWidget.groupId != null) {
              final PathProviderFoundation provider = PathProviderFoundation();
              directory = await provider.getContainerPath(
                appGroupIdentifier: HomeWidget.groupId!,
              );
            } else {
              throw UnsupportedError(
                'No groupId defined. Did you forget to call `HomeWidget.setAppGroupId`',
              );
            }
          } else {
            directory = (await getApplicationSupportDirectory()).path;
          }
```

Reply:
I think I prefer the assertion to check. Also I guess the whole try/catch for the `UnsupportedError` could be omitted  as we are now checking for `isIOS` which is the only platform where `getContainerPath` is supported per https://pub.dev/documentation/path_provider_foundation/latest/path_provider_foundation/PathProviderFoundation/getContainerPath.html so this is actually not needed anymore

Reply:
```dart
        if (Platform.isIOS) {
          final PathProviderFoundation provider = PathProviderFoundation();
          assert(
            HomeWidget.groupId != null,
            'No groupId defined. Did you forget to call `HomeWidget.setAppGroupId`',
          );
          directory = await provider.getContainerPath(
            appGroupIdentifier: HomeWidget.groupId!,
          );
        } else {
          directory = (await getApplicationSupportDirectory()).path;
        }
```

please check if this requires any changes. Also removed the parent try catch for the same unsupported error.

Reply:
Yes this looks good 👍 If you could update this PR I think I should be able to get this landed in a new version this week

Reply:
done